File: src/python/PythonSDK/foundationallm/models/orchestration/internal_context_completion_request.py
"""
Class Name: InternalContextCompletionRequest
Description: Encapsulates the metadata required to complete an internal context orchestration request.
"""
from typing import Optional
from foundationallm.models.metadata import InternalContextAgent
from .completion_request_base import CompletionRequestBase

class InternalContextCompletionRequest(CompletionRequestBase):
    """
    Internal Context completion request.
    """    
    agent: Optional[InternalContextAgent] = None

File: src/dotnet/Common/Models/Orchestration/InternalContextCompletionRequest.cs
using FoundationaLLM.Common.Models.Agents;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// The Internal Context Completion Request model.
    /// </summary>
    public class InternalContextCompletionRequest : LLMCompletionRequest
    {
        /// <summary>
        /// The Internal Context agent metadata.
        /// </summary>
        [JsonPropertyName("agent")]
        public required InternalContextAgent Agent { get; set; }
    }
}
